src/lib.rs

```diff
@@ -551,7 +568,7 @@ impl Header {
     }
 
     /// Returns a view into this header as a byte array.
-    fn as_bytes(&self) -> &[u8; 512] {
+    pub fn as_bytes(&self) -> &[u8; 512] {
         unsafe { &*(self as *const _ as *const [u8; 512]) }
```

Comment:
Could you also add a `debug_assert_eq!(512, mem::size_of_val(self))` here? Probably should have added that initially as well but if it's publicly exposed it's probably more important